src/db.ts

```diff
@@ -3,9 +3,14 @@ import { MONGO_DB_URL } from "./constants";
 import { logger } from "./libraries";
 import { checkReplicaSet } from "./utilities/checkReplicaSet";
 
+let isConnected = false;
 let session!: mongoose.ClientSession;
 
 export const connect = async (dbName?: string): Promise<void> => {
+  if (isConnected) {
+    return;
+  }
+
```

Comment:
hey @lokeshwar777 creating a separate variable for checking connection state can't be a good practice, it could cause weird behaviour. Use something like `mongoose.connection.readyState` in the files that are calling the `connect()` functions.

Reply:
@meetulr Thanks for the feedback, I have made the changes you have suggested.